pkg/clean/clean.go

```diff
@@ -0,0 +1,84 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package clean
+
+import (
+	"flag"
+	"fmt"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/securitypolicy"
+)
+
+var (
+	log          = logger.Log
+	cf           *config.NSXOperatorConfig
+	mgrIp        string
+	vcEndpoint   string
+	vcUser       string
+	vcPasswd     string
+	vcSsoDomain  string
+	vcHttpsPort  int
+	thumbprint   string
+	vmcaCertFile string
+	cluster      string
+)
+
+func init() {
+	flag.StringVar(&vcEndpoint, "vc-endpoint", "", "nsx manager ip")
+	flag.StringVar(&vcSsoDomain, "vc-sso-domain", "", "nsx manager ip")
+	flag.StringVar(&mgrIp, "mgr-ip", "", "nsx manager ip")
+	flag.StringVar(&vcUser, "vc-user", "", "nsx username")
+	flag.StringVar(&vcPasswd, "vc-passwd", "", "nsx password")
+	flag.IntVar(&vcHttpsPort, "vc-https-port", 443, "nsx password")
+	flag.StringVar(&thumbprint, "thumbprint", "", "nsx thumbprint")
+	flag.StringVar(&vmcaCertFile, "vmca-cert-file", "", "vmca cert file")
+	flag.StringVar(&cluster, "cluster", "", "cluster name")
+	flag.IntVar(&config.LogLevel, "log-level", 0, "Use zap-core log system.")
+	flag.Parse()
+
+	logf.SetLogger(logger.ZapLogger())
+	cf = config.NewNSXOpertorConfig()
+	cf.NsxApiManagers = []string{mgrIp}
+	cf.VCUser = vcUser
+	cf.VCPassword = vcPasswd
+	cf.VCEndPoint = vcEndpoint
+	cf.SsoDomain = vcSsoDomain
+	cf.HttpsPort = vcHttpsPort
+	cf.Thumbprint = []string{thumbprint}
+	cf.NsxApiCertFile = vmcaCertFile
+	cf.Cluster = cluster
+}
+
+func Clean() error {
+	log.Info("starting NSX cleanup")
+	var err error
+	var securityPolicyService *securitypolicy.SecurityPolicyService
+
+	nsxClient := nsx.GetClient(cf)
+	if nsxClient == nil {
+		return fmt.Errorf("failed to get nsx client")
+	}
+
+	var commonService = common.Service{
+		NSXClient: nsxClient,
+		NSXConfig: cf,
+	}
+
+	if securityPolicyService, err = securitypolicy.InitializeSecurityPolicy(commonService); err != nil {
+		return fmt.Errorf("failed to initialize securitypolicy service: %v", err)
+	}
+	err = securityPolicyService.Cleanup()
+	if err != nil {
+		return fmt.Errorf("failed to clean up: %v", err)
```

Comment:
nit: Use the `%w` directive instead of `%v`. This will wrap the error instead, meaning we retain the details of the original error.

Reply:
ok